lib/core/plugins/pipeRunner.js

```diff
@@ -0,0 +1,101 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2018 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const
+  assert = require('assert'),
+  Denque = require('denque'),
+  async = require('async'),
+  {errors: {KuzzleError}} = require('kuzzle-common-objects'),
+  errorsManager = require('../../util/errors').wrap('plugin', 'runtime');
+
+/**
+ * @private
+ * @class PipeChain
+ */
+class PipeChain {
```

Comment:
Does this class declaration have any impact on v8 performances or it's just for readability and maintainability?

Reply:
Both: declaring a class and not addind new properties to it on the fly allows turbofan to convert it to a C++ class with direct access (i.e. 0 cost) to its properties.

And it's also more readable to manipulate class instances for stored objects, because when doing maintenance you don't have to spend time wondering what properties that object might have